packages/lesswrong/lib/collections/comments/newSchema.ts

```diff
@@ -771,6 +785,26 @@ const schema = {
       },
     },
   },
+  // draft: Indicates whether a comment is a draft.
+  // Draft comments are only visible to authors and admins.
+  draft: {
+    database: {
+      type: "BOOL",
+      defaultValue: false,
+      canAutofillDefault: true,
+      nullable: false,
+    },
+    graphql: {
+      outputType: "Boolean!",
+      inputType: "Boolean",
+      canRead: ["guests"], // Access to the whole comment is gated in `commentCheckAccess`, not here
+      canUpdate: [userOwns, "sunshineRegiment", "admins"],
```

Comment:
I haven't totally finished reviewing the PR but so far it doesn't look like you intend to allow users to set their comments back to draft after they've been published, so maybe that also wants a check in `mutations`/`commentCallbackFunctions`?